### DISCUSSION TITLE
C# Markup - Material TabBar Style

### DISCUSSION THREAD

**Discussion by JohnWimbish** [unoplatform/uno#14827] [Q&A] [answered]
About halfway down this page: 
https://platform.uno/docs/articles/external/uno.toolkit.ui/doc/controls/TabBarAndTabBarItem.html#top-tabbar-style

Is a discussion of pre-built styles, with this code:
```
xmlns:utu="using:Uno.Toolkit.UI"

<utu:TabBar SelectedIndex="1"
            Style="{StaticResource ColoredTopTabBarStyle}">
    <utu:TabBar.Items>
        <utu:TabBarItem Content="Home" />
        <utu:TabBarItem Content="Search" />
        <utu:TabBarItem Content="Support" />
        <utu:TabBarItem Content="About" />
    </utu:TabBar.Items>
</utu:TabBar>
```

I am trying, in C#, to set the style to ColoredTopTabBarStyle. 
```
new TabBar()
    .Style( ***What Goes Here?*** )
    .Items(
        ... 
    ),
```
How do I reference ColoredTopTabBarStyle and TopTabBarStyle so as to set it in C#? I can't find them via Intellisense.
Thanks!

**Comment by kazo0** [comment, marked as answer]
Hi @JohnWimbish!

We are currently in the process of updating the Themes and Toolkit sections of the docs to include C# Markup examples. 

For this case, you can either use the name of the style key directly as part of a StaticResource using the DependencyPropertyBuilder extensions, outlined [here](https://platform.uno/docs/articles/external/uno.extensions/doc/Overview/Markup/StaticAndThemeResources.html).

So it would look something like:

```csharp
new TabBar()
    .Style(x => x.StaticResource("ColoredTopTabBarStyle"))
    .Items(
        ... 
    ),
```

Or, more excitingly, the Uno Toolkit and the accompanying Uno Toolkit Material library (that contains the Material ColoredTopTabBarStyle), both have C# Markup support in the form of an additional NuGet package that can be found in the [Toolkit Getting Started - C# Markup section](https://platform.uno/docs/articles/external/uno.toolkit.ui/doc/material-getting-started.html#using-c-markup).

So, with the Uno Toolkit Markup package, it provides strongly-typed resource key properties. Your code would then look like this:

```csharp
new TabBar()
    .Style(Theme.TabBar.Styles.ColoredTop)
    .Items(
        ... 
    ),
```

The static property can be found here: https://github.com/unoplatform/uno.toolkit.ui/blob/1874702add9647e9c2ac4f852cd8ba623df400bc/src/library/Uno.Toolkit.WinUI.Markup/Theme/TabBar.cs#L131

**Reply by JohnWimbish** [reply]
Brilliant, @kazo0! Tried them both, with success. 
I really like where you all are going with the Theme.TabBar.Styles.ColoredTop option!
